cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,502 @@
+    current_checkset = ''
+    for check, meta in checks.items():
+        # Check if the purpose has changed - if so create a new
+        # section title:
+        if meta['purpose'] != current_checkset:
+            current_checkset = meta['purpose']
+            title = CHECKS_DESC[meta["purpose"]]
+            output += f'\n{title}\n{"-" * len(title)}\n\n'
+
+        # Fill a template with info about the issue.
+        template = (
+            '{checkset}{index:003d}:\n    {summary}\n\n'
+        )
+        if meta['url'].startswith('http'):
+            url = meta['url']
+        else:
+            url = URL_STUB + meta['url']
+        msg = template.format(
+            title=check.pattern.replace('\\', ''),
+            checkset=meta['purpose'],
+            summary=meta['short'],
+            url=url,
+            index=meta['index'],
+        )
+        output += msg
+    output += '\n'
+    return output
+
+
+def get_option_parser() -> COP:
+    parser = COP(
+        COP_DOC,
+        argdoc=[COP.optional(('DIR ...', 'Directories to lint'))],
+    )
+    parser.add_option(
+        '--inplace', '-i',
+        help=(
+            'Modify files in place, adding comments to files. '
+            'If not set, the script will work as a linter'
+        ),
+        action='store_true',
+        default=False,
+    )
+    parser.add_option(
+        '--ruleset', '-r',
+        help=(
+            'Set of rules to use: '
+            '("728", "style", "all")'
+        ),
+        default='728',
+        choices=('728', 'style', 'all'),
+        dest='linter'
+    )
+    parser.add_option(
+        '--reference', '--ref', '-R',
+        help=(
+            'Print Reference for error codes.'
+        ),
+        action='store_true',
+        default=False,
+        dest='ref_mode'
+    )
+
+    return parser
+
+
+@cli_function(get_option_parser)
+def main(parser: COP, options: 'Values', *targets) -> None:
+
+    if options.ref_mode:
+        print(get_reference_text(parse_checks(options.linter)))
+        exit(0)
+
+    # Expand paths so that message will return full path
+    # & ensure that CWD is used if no path is given:
+    if targets:
+        targets = tuple(Path(path).resolve() for path in targets)
+    else:
+        targets = (str(Path.cwd()),)
+
+    # Get a list of checks bas ed on the checking options:
+    count = 0
+    # Allow us to check any number of folders at once
+    for target in targets:
+        target = Path(target)
+        if not target.exists():
+            LOG.warn(f'Path {target} does not exist.')
+        else:
+            # Check whether target is an upgraded Cylc 8 workflow.
+            # If it isn't then we shouldn't run the 7-to-8 checks upon
+            # it:
+            cylc8 = (target / 'flow.cylc').exists()
+            if not cylc8 and options.linter == '728':
+                LOG.error(
+                    f'{target} not a Cylc 8 workflow: '
+                    'No checks will be made.'
```

Comment:
```suggestion
                    Convert** top level `suite.rc` '
                    'to `flow.cylc` to run checks.
```

Reply:
I feel like that's a bit too active. We only want them to rename when they're ready.
```suggestion
                    'Lint after renaming '
                    '"suite.rc" to "flow.cylc"'
```